### PR TITLE
fix(Auto-fix): ignore is not applied

### PR DESCRIPTION
**Pull request by taro-28**
Closes https://github.com/webpro-nl/knip/issues/653

Fixed and added test cases for the following cases

- `ignore`.
- ignoreDependencies`
- use `ignoreWorkspaces` for ignore
- `tags` for ignore

Since `ignoreMembers` is not yet supported by Auto-fix and I thought `ignoreBinaries` was an option unrelated to Auto-fix, I did not include either of them in the test case.

### CODE REVIEW

**Comment by taro-28** [comment]
Sorry, I'll check the CI error.

**Comment by taro-28** [comment]
> Sorry, I'll check the CI error.

I checked and could not reproduce it locally.
It is probably a falky test and seems to be unrelated to the changes in this PR.

File: packages/knip/src/index.ts [hunk 1]
```diff
@@ -519,8 +519,10 @@ export const main = async (unresolvedConfiguration: CommandLineOptions) => {
                 line: exportedItem.line,
                 col: exportedItem.col,
               });
-              if (isType) fixer.addUnusedTypeNode(filePath, exportedItem.fixes);
-              else fixer.addUnusedExportNode(filePath, exportedItem.fixes);
+              if (issue) {
```
Comment (webpro) [inline packages/knip/src/index.ts:522]: Nitpick: since it's a boolean I'd like to name it `isAdded` or `isIssueAdded`

**Review by webpro — APPROVED** [review]
Thanks a lot for this PR! Just a nitpick and I'm happy to merge 🙌

File: packages/knip/src/index.ts (hunk 1 above)
Comment (taro-28) [reply packages/knip/src/index.ts]: Indeed! I fixed it to `isIssueAdded`
https://github.com/webpro-nl/knip/pull/654/commits/fffce7476772c824de42e04f7b99685db949e77d

**Comment by webpro** [comment]
Thank you @taro-28, most excellent stuff!

**Comment by webpro** [comment]
:rocket: _This pull request is included in v5.17.0-canary.2. See [Release 5.17.0-canary.2](https://github.com/webpro-nl/knip/releases/tag/5.17.0-canary.2) for release notes._

_Using Knip in a commercial project? Please consider [sponsoring me](https://github.com/sponsors/webpro)._

**Pull request merged**